pkg/reposync/conditions_test.go

```diff
@@ -239,79 +274,263 @@ func TestClearCondition(t *testing.T) {
 		})
 	}
 }
 
 func TestSetStalled(t *testing.T) {
+	now = func() metav1.Time {
+		return initialNow
+	}
 	testCases := []struct {
-		name   string
-		rs     *v1beta1.RepoSync
-		reason string
-		err    error
-		want   []v1beta1.RepoSyncCondition
+		name             string
+		rs               *v1beta1.RepoSync
+		reason           string
+		err              error
+		want             []v1beta1.RepoSyncCondition
+		wantUpdated      bool
+		wantTransitioned bool
 	}{
 		{
 			"Set new stalled condition",
 			fake.RepoSyncObjectV1Beta1(testNs, configsync.RepoSyncName),
 			"Error1",
 			errors.New("this is error 1"),
 			[]v1beta1.RepoSyncCondition{
-				fakeCondition(v1beta1.RepoSyncStalled, metav1.ConditionTrue, "Error1", "this is error 1"),
+				// Update and transition
+				fakeCondition(v1beta1.RepoSyncStalled, metav1.ConditionTrue, updatedNow, updatedNow, "Error1", "this is error 1"),
 			},
+			true,
+			true,
 		},
 		{
 			"Update existing stalled condition",
-			fake.RepoSyncObjectV1Beta1(testNs, configsync.RepoSyncName, withConditions(fakeCondition(v1beta1.RepoSyncReconciling, metav1.ConditionTrue), fakeCondition(v1beta1.RepoSyncStalled, metav1.ConditionFalse))),
+			fake.RepoSyncObjectV1Beta1(testNs, configsync.RepoSyncName,
+				withConditions(
+					fakeCondition(v1beta1.RepoSyncReconciling, metav1.ConditionTrue, initialNow, initialNow),
+					fakeCondition(v1beta1.RepoSyncStalled, metav1.ConditionFalse, initialNow, initialNow))),
 			"Error2",
 			errors.New("this is error 2"),
 			[]v1beta1.RepoSyncCondition{
-				fakeCondition(v1beta1.RepoSyncReconciling, metav1.ConditionTrue),
-				fakeCondition(v1beta1.RepoSyncStalled, metav1.ConditionTrue, "Error2", "this is error 2"),
+				// No update or transition
+				fakeCondition(v1beta1.RepoSyncReconciling, metav1.ConditionTrue, initialNow, initialNow),
+				// Update and transition
+				fakeCondition(v1beta1.RepoSyncStalled, metav1.ConditionTrue, updatedNow, updatedNow, "Error2", "this is error 2"),
 			},
+			true,
+			true,
+		},
+	}
+	now = func() metav1.Time {
+		return updatedNow
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			updated, transitioned := SetStalled(tc.rs, tc.reason, tc.err)
+			if diff := cmp.Diff(tc.want, tc.rs.Status.Conditions); diff != "" {
+				t.Error(diff)
+			}
+			assert.Equal(t, tc.wantUpdated, updated, "updated")
+			assert.Equal(t, tc.wantTransitioned, transitioned, "transitioned")
+		})
+	}
+}
+
+func TestSetSyncing(t *testing.T) {
+	testCases := []struct {
+		name             string
+		rs               *v1beta1.RepoSync
+		status           bool
+		reason           string
+		message          string
+		commit           string
+		errorSources     []v1beta1.ErrorSource
+		errorSummary     *v1beta1.ErrorSummary
+		timestamp        metav1.Time
+		want             []v1beta1.RepoSyncCondition
+		wantUpdated      bool
+		wantTransitioned bool
+	}{
+		{
+			"Set new syncing condition without error",
```

Comment:
Could you use named fields in the assignments? It's a bit hard to follow with so many parameters

Reply:
None of the others in the file use named fields... I was just being consistent. But I also hate this pattern.

Reply:
I'm not suggesting you need to update everything else, just to use it for the new tests. But it's not a big deal, we can refactor it in a subsequent change.

Reply:
Inserted them all with a neat vscode-go feature: https://github.com/golang/vscode-go/blob/master/docs/settings.md#goinlayhintscompositeliteralfields